Class refactoring

TheGreatRefrigerator opened the pull request.

TheGreatRefrigerator commented on src/OrsDirections.js:
```diff
@@ -6,6 +6,16 @@ import OrsBase from './OrsBase'
 const orsUtil = new OrsUtil()
 
 class OrsDirections extends OrsBase {
+  constructor(args) {
+    super(args);
+    if (!this.defaultArgs[Constants.propNames.service]) {
+      this.defaultArgs[Constants.propNames.service] = 'directions'
+    }
+    if (!args[Constants.propNames.apiVersion]) {
+      this.defaultArgs.api_version = Constants.defaultAPIVersion
+    }
+  }
```
I need to think about this bit a little longer

TheGreatRefrigerator reviewed (commented):
@EmmaHegarty 
refactor topUpArgs() to fillArgs() can be squashed to the commit earlier where you already modify this function.

Some of the tests also need fixing with these changes.

Hmm should have let you do the PR, now i can't request changes on my own PR :)

TheGreatRefrigerator commented on src/OrsMatrix.js:
```diff
@@ -7,22 +7,23 @@ const orsUtil = new OrsUtil()
     }
 
-    orsUtil.copyProperties(reqArgs, this.defaultArgs)
+    this.requestArgs = orsUtil.topUpArgs(this.defaultArgs,this.requestArgs)
```
here as well

TheGreatRefrigerator commented on src/OrsDirections.js:
```diff
@@ -51,21 +51,22 @@ class OrsDirections extends OrsBase {
       this.defaultArgs[Constants.propNames.service] = 'directions'
     }
-    orsUtil.copyProperties(reqArgs, this.defaultArgs)
+    this.requestArgs = orsUtil.topUpArgs(this.defaultArgs,this.requestArgs)
```
there is still topUpArgs here

TheGreatRefrigerator commented on examples/directions-two-in-one.html:
```diff
@@ -0,0 +1,101 @@
+<!DOCTYPE html>
```
Was this relevant for the branch?
Otherwise it shouldn't be in the refactor branch.
I would move it to another local branch and add after PR.

EmmaHegarty replied:
The file was used to manually test if different metadata is created for multiple requests in one file, before cypress testing was implemented.
I have removed it and written a cypress test instead.

TheGreatRefrigerator commented on src/__tests__/OrsUtil.cy.js:
```diff
@@ -0,0 +1,81 @@
+
+  it('properties are removed', () => {
+    let response = orsUtil.
+      prepareRequest({})
```
This test does not really test the function, as it runs on an empty object.
We need an object with properties here that are actually removed.

TheGreatRefrigerator reviewed (commented):
Still a bit left to do, but i'm not sure if you pushed your edits from last session?
I thought we already took care of some of the issues...

TheGreatRefrigerator commented:
@EmmaHegarty I've added some final adaptions:
- moved OrsUtils.setRequestDefaults to OrsBase._setRequestDefaults as private method as it is only needed there
- splitted the tests for this method into a separate commit
- fixed formatting of the Directions test file (remind me to look into the autoformatting of your IDE)
- other small fixes
